microsoft/winget-cli
# WinGet COM API for Repair

This PR introduces the COM API for WinGet Repair Operation. Key updates include:
- PackageManager.IDL updates for Repair API contracts.
- Implementation of Repair COM API methods in PackageManager runtime class.
- Addition of runtime classes for RepairOptions and RepairResults.
- Integration logic connecting COM API with existing Repair Workflow.
- Updates to WinGetCLIPackage manifest for Repair COM API GUIDs.
- Changes to Microsoft.Management.Deployment.InProc to support E2E testing of Repair COM API interops.

Add Repair-WinGetPackage powershell cmdlet to Microsoft.WinGet.Client
- Introduced a new cmdlet `Repair-WinGetPackage` to the Microsoft.WinGet.Client PowerShell module

**[Tests:]**
Added E2E tests for Winget COM Repair API:
- Introduced RepairInterop.cs for InProc & Out of Proc COM Repair APIs.
- Extended GroupPolicyForInterop.cs to cover COM Repair API scenarios.

Added PowerShell pester tests for Repair-WinGetPackage cmdlet.


**[Validation:]**
- Executed RepairInterop E2E tests and confirmed all tests passed.
- Executed GroupPolicyForInterop E2E tests and confirmed all tests passed.
- Executed RepairCommand E2E tests and confirmed all tests passed.

![image](https://github.com/user-attachments/assets/b43084a8-e3ae-41fe-bb91-f125f792f496)

- Executed the Repair-WinGetPackage PowerShell cmdlet manually for a set of test packages and confirmed that it returns the expected results. ex:
```
- AppInstallerTest.TestMsixInstaller
  - Install-WinGetPackage -Id AppInstallerTest.TestMsixInstaller - return status OK
  - Repair-WinGetPackage -Id AppInstallerTest.TestMsixInstaller - return status OK 
  - Uninstall-WinGetPackage -Id AppInstallerTest.TestMsixInstaller - return status OK
- AppInstallerTest.TestModifyRepair
  - Install-WinGetPackage -Id AppInstallerTest.TestModifyRepair - return status OK
  - Repair-WinGetPackage -Id AppInstallerTest.TestModifyRepair - return status OK
  - Uninstall-WinGetPackage -Id AppInstallerTest.TestModifyRepair - return status OK
```




- [x] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [x] This pull request is related to an issue.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4736)
